fastapi_decorators/decorators.py

```diff
@@ -57,15 +57,34 @@ def update_user(*, user_id: int, user_update: UserUpdate):
     """
 
     def decorator(func: F) -> F:
-        original_signature = signature(func)
-        original_parameters = original_signature.parameters
+        original_func = unwrap(func)
+        original_signature = signature(func, follow_wrapped=True)
+
+        hints = get_type_hints(
+            original_func,
+            globalns=original_func.__globals__,
+            localns=None,
+        )
+
+        resolved_params: dict[str, Parameter] = {}
+        for name, param in original_signature.parameters.items():
+            anno = hints.get(name, param.annotation)
+            resolved_params[name] = param.replace(annotation=anno)
+
+        dep_params = _add_dependency_parameters(
+            args,
+            kwargs,
+            MappingProxyType(resolved_params),
+        )
+
+        all_params = {**resolved_params, **dep_params}
```

Comment:
@Poogles This seems to work, and as far as I can tell we're not risking much in terms of side effects.
We're effectively resolving the forward refs explicitly and rebuilding the signature.

Any thoughts or concerns with this?